src/Properties/properties.cpp

```diff
@@ -38,4 +39,114 @@ Vector<double, 3> calculateElectronicDipoleMoment(const VectorSQOneElectronOpera
+        throw std::runtime_error("properties::calculateDysonOrbital(WaveFunction, WaveFunction): wavefunction2 is not in a product Fock space");
+    }
+
+    const auto fock_space1 = static_cast<const ProductFockSpace&>(wavefunction1.get_fock_space());
+    const auto fock_space2 = static_cast<const ProductFockSpace&>(wavefunction2.get_fock_space());
+
+    if (!((fock_space1.get_N_alpha() - fock_space2.get_N_alpha() == 0) && (fock_space1.get_N_beta() - fock_space2.get_N_beta() == 1))) {
+        if (!((fock_space1.get_N_alpha() - fock_space2.get_N_alpha() == 1) && (fock_space1.get_N_beta() - fock_space2.get_N_beta() == 0))) {
+            throw std::runtime_error("properties::calculateDysonOrbital(WaveFunction, WaveFunction): wavefunction2 is not expressed in a product Fock space with one fewer electron than wavefunction1");
+        }
+    } 
+
+    const auto fock_space_alpha1 = fock_space1.get_fock_space_alpha();
+    const auto fock_space_alpha2 = fock_space2.get_fock_space_alpha();
+    const auto fock_space_beta1 = fock_space1.get_fock_space_beta();
+    const auto fock_space_beta2 = fock_space2.get_fock_space_beta();
+
+    const auto ci_coeff1 = wavefunction1.get_coefficients();
+    const auto ci_coeff2 = wavefunction2.get_coefficients();
+
+    VectorX<double> dyson_coeff = VectorX<double>::Zero(fock_space1.get_K());
+    size_t dim_alpha = fock_space_alpha1.get_dimension();
+    size_t dim_beta = fock_space_beta1.get_dimension();
+
+    // Given the one electron difference requirement, one of the spin Fock spaces will be identical for both wave functions, identifying that spin Fock space allows for a simple algorithm
+
+    // Beta electron differs by one
+    if ((fock_space1.get_N_alpha() - fock_space2.get_N_alpha() == 0) && (fock_space1.get_N_beta() - fock_space2.get_N_beta() == 1)) {
+
+        ONV onv_beta = fock_space_beta1.makeONV(0);
+
+        for (size_t Ib = 0; Ib < dim_beta; Ib++) {  // Ib loops over addresses of beta spin strings
+            int sign = -1;
+            for (size_t e_b = 0; e_b < fock_space_beta1.get_N(); e_b++) {  // loop over beta electrons
+                sign *= -1;
+                size_t p = onv_beta.get_occupation_index(e_b);
+
+                onv_beta.annihilate(p);
+
+                size_t address = fock_space_beta2.getAddress(onv_beta.get_unsigned_representation());
+
+                onv_beta.create(p);
+
+                double coeff = 0;
+
+                for (size_t Ia = 0; Ia < dim_alpha; Ia++) {  // alpha Fock space is identical
+                    coeff += sign * ci_coeff1(Ia * dim_beta + Ib) * ci_coeff2(address * dim_beta + Ib);
+                }
+
+                dyson_coeff(p) += coeff;
+            }
+
+
+            if (Ib < dim_beta - 1) {  // prevent last permutation to occur
+                fock_space_beta1.setNextONV(onv_beta);
+            }
+        }  // beta address (Ib) loop
+    }
+
+    // Alpha electrons differs by one
```

Comment:
idem: explain

Reply:
This seems like a copy-paste: is there a way you can remove some duplicate codes?

Reply:
You are correct, I have modified this, however it does look a tad more complicated